typings/index.d.ts

```diff
@@ -0,0 +1,410 @@
+import {RequireAtLeastOne} from './type-utils'
```

Comment:
This needs to be broken down into separate files for each type.

This file is too large.

Reply:
Done